src/persist/src/operators/upsert.rs

```diff
@@ -128,11 +128,12 @@ where
             let snapshot = persist_config.read_handle.snapshot();
             let (restored_oks, restored_errs) =
                 self.scope().replay(snapshot).ok_err(|x| split_ok_err(x));
-            let restored_upsert_oks = restored_oks.filter_and_retract_future_updates(
-                name,
-                persist_config.write_handle.clone(),
-                persist_config.upper_seal_ts,
-            );
+            let (restored_upsert_oks, _retract_errs) = restored_oks
```

Comment:
i'm not a big fan of throwing away errors. it's unclear to me (as it is to everyone) what our error reporting story is, but is there some good reason we're not concat'ing them in here?

Reply:
I don't know what I was thinking here. :see_no_evil: In my other PR I'm expressly wiring through more error streams: https://github.com/MaterializeInc/materialize/pull/9246

Pushing a fix...